solidity/contracts/BancorExchange.sol

```diff
@@ -55,12 +56,27 @@ contract BancorExchange is Owned {
         quickBuyPath = _path;
     }
 
-    function buyRING(uint _minReturn) payable returns (uint){
+    function buyRING(uint _minReturn) payable returns (uint) {
         uint amount = bancorConverter.quickConvert.value(msg.value)(quickBuyPath, msg.value, _minReturn);
         smartToken.transfer(msg.sender, amount);
         return amount;
     }
 
+    function exchangeRingInBid(uint _minReturn, address _buyer) returns (uint) {
+        require(msg.sender == clockAuction);
+        uint connectorBalance = bancorConverter.getConnectorBalance(quickBuyPath[0]);
+        uint supply = smartToken.totalSupply();
+        uint cw = bancorConverter.connectors(quickBuyPath[0]).weight;
+        uint amountRequired = bancorConverter.getPurchaseRequire(connectorBalance, supply, cw, _minReturn);
+        require(msg.value >= amountRequired);
+        uint amount = bancorConverter.quickConvert.value(amountRequired)(quickBuyPath, msg.value, _minReturn);
```

Comment:
Here we have msg.value, but this method is not payable, it seems that something will go wroing?